WordPress/Classes/ViewRelated/Reader/ReaderSiteStreamHeader.swift

```diff
@@ -29,25 +34,74 @@ import Foundation
         WPStyleGuide.applyReaderSiteStreamCountStyle(followCountLabel)
     }
 
+    public override func sizeThatFits(size: CGSize) -> CGSize {
+        // Vertical and horizontal margins
+        let hMargin = descriptionLabel.frame.minX
+        let vMargin = descriptionLabel.frame.minY
+
+        let width = UIDevice.isPad() ? contentIPadWidthConstraint!.constant : size.width
+        let innerWidth = width - (hMargin * 2)
+        let adjustedSize = CGSize(width:innerWidth, height:CGFloat.max)
+
+        var height = UIDevice.isPad() ? contentIPadTopConstraint!.constant : 0
+        height += descriptionView.frame.minY
+        height += vMargin
+        height += descriptionLabel.sizeThatFits(adjustedSize).height
+        height += descriptionBottomConstraint.constant
+        height += followCountLabel.sizeThatFits(adjustedSize).height
+        height += followCountBottomConstraint.constant;
+
+        return CGSize(width: size.width, height: height)
+    }
+
 
    // MARK: - Configuration
 
     public func configureHeader(topic: ReaderAbstractTopic) {
-        // TODO: Wire up actual display when supported in core data
-        avatarImageView.setImageWithURL(NSURL(), placeholderImage: UIImage(named: defaultBlavatar))
-        titleLabel.text = topic.title
-        detailLabel.text = "site.com"
-        if topic.following {
+        assert(topic.isKindOfClass(ReaderSiteTopic), "Topic must be a site topic")
+
+        let siteTopic = topic as! ReaderSiteTopic
```

Comment:
I'm sure there's a Swiftier way to declare that protocol with associated types, but I can't figure it out :wink: